update string/match (#54)
OI-wiki/i18n: docs/string/match.md

## String matching problem

### Single string match

> There are a pattern string (pattern), and a string to be matched. We need to find all occurrences of the former in the latter.

Reviewer: ```suggestion
There is a pattern string, and a string to be matched. We need to find all occurrences of the former string in the latter string.
```

### Multi-string matching

> There are multiple pattern strings, and one string to be matched (multiple strings to be matched can be directly connected).

Reviewer: concatenated

> It is certainly possible to treat it as a single string matching problem, but the efficiency is not good enough.

Reviewer: ```suggestion
It is certainly possible to treat it as multiple single string matching problems, but the efficiency is not good enough.
```

> ### Matches any suffix of a string

Reviewer: ```suggestion
### Match any suffix of a string
```

Reviewer: 这里 upstream 有 [refactor](https://github.com/OI-wiki/OI-wiki/pull/2495)，到时候看一下上游合了这里也改一下吧
